### PR TITLE
[BREAKINGCHANGE] change legend position values to PascalCase, fix resize glitch

### PR DESCRIPTION
**Pull request by sjcobb**
This PR includes:
- [BREAKINGCHANGE] change legend position values to PascalCase
- [BUGFIX] chart resize glitch when changing legend position 

The `legend.position` change to use PascalCase for dashboard values that are string unions/enums was made for consistency with other properties like `kind` and `calculation`. I also took a quick stab at FE validation for the legend spec when inside the panel edit drawer, but this isn't meant to be a long term solution.

See Matrix chat discussion about snake_case for keys in dashboard JSON and PascalCase for values: 
https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$1yAfafCLxONIoBfj83jN0ul2v8AhHBPXpG2QZfwHJUk?via=matrix.org&via=herail.net

### CODE REVIEW

File: ui/components/src/LegendOptionsEditor/LegendOptionsEditor.test.tsx [hunk 1]
```diff
@@ -29,10 +29,27 @@ describe('LegendOptionsEditor', () => {
...
   };
 
+  const getLegendPositionSelector = () => {
+    return screen.getByRole('combobox', { name: 'Position' });
+  };
+
   it('can change legend visibility by clicking', () => {
     const onChange = jest.fn();
     renderLegendOptionsEditor(undefined, onChange);
+    expect(getLegendPositionSelector()).toBeDisabled();
```
Comment (sjcobb) [inline ui/components/src/LegendOptionsEditor/LegendOptionsEditor.test.tsx:39]: Followed up on TODOs from #845 (see @juliepagano [thread here](https://github.com/perses/perses/pull/845#discussion_r1042715274))

**Comment by Nexucis** [comment]
I just added a short fix in the `mig.cuepart` file @sjcobb . Tests are green now :)

**Review by Nexucis — APPROVED** [review]

File: ui/components/src/model/legend.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { MouseEventHandler } from 'react';
+
+export const legendPositions = ['Bottom', 'Right'] as const;
+
+export type LegendPositions = typeof legendPositions[number];
```
Comment (eunicorn) [inline ui/components/src/model/legend.ts:18]: is it simpler if we just do `type LegendPositions = 'Bottom' | 'Right'`?

File: ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx [hunk 3]
```diff
@@ -47,7 +48,10 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
   } = props;
 
   // popuate default 'position' and other future properties
```
Comment (eunicorn) [inline ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx:50]: typo?

**Review by eunicorn — APPROVED** [review]

File: ui/components/src/model/legend.ts (hunk 2 above)
Comment (sjcobb) [reply ui/components/src/model/legend.ts:18]: So I ended up with it this way so I could use legendPositions further down in isValidLegendPosition: https://github.com/perses/perses/blob/d8cf60b09b5965eeb17773706f25efd390829ad8/ui/components/src/model/legend.ts#L55

**Pull request merged**